src/subcommand.rs

```diff
@@ -194,44 +142,31 @@ impl Subcommand {
     }
 
     pub fn quiet(&self) -> bool {
-        self.quiet
+        self.args.quiet
     }
 
     pub fn config(&self) -> Option<&LocalizedConfig> {
         self.config.as_ref()
     }
-}
 
-#[cfg(test)]
-mod tests {
-    use super::*;
-
-    #[test]
-    fn test_separator_space() {
-        let args = [
-            "cargo",
-            "subcommand",
-            "build",
-            "--target",
-            "x86_64-unknown-linux-gnu",
-        ]
-        .iter()
-        .map(|s| s.to_string());
-        let cmd = Subcommand::new(args, "subcommand", |_, _| Ok(false)).unwrap();
-        assert_eq!(cmd.target(), Some("x86_64-unknown-linux-gnu"));
+    pub fn build_dir(&self, target: Option<&str>) -> PathBuf {
+        let target_dir = dunce::simplified(self.target_dir()).to_path_buf();
+        let arch_dir = if let Some(target) = target {
+            target_dir.join(target)
+        } else {
+            target_dir
```

Comment:
Nit: You can postpone the "cloning" of `&Path` only when not needing an extra allocation in `.join()`:

```suggestion
        let target_dir = dunce::simplified(self.target_dir());
        let arch_dir = if let Some(target) = target {
            target_dir.join(target)
        } else {
            target_dir.to_path_buf()
```